### ISSUE TITLE
docker ps --quiet doesn't work with psFormat table

### ISSUE THREAD

**Issue by stayclassychicago** [docker/docker#15673]
Looks like docker ps --quiet doesn't work when using new psFormat options, introduced in docker 1.8
https://github.com/docker/docker/issues/7477

docker version:
```
Client:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Thu Aug 13 19:47:52 UTC 2015
 OS/Arch:      darwin/amd64

Server:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Thu Aug 13 02:49:29 UTC 2015
 OS/Arch:      linux/amd64
```

docker info:
```
Containers: 1
Images: 142
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 144
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.0.9-boot2docker
Operating System: Boot2Docker 1.8.1 (TCL 6.3); master : 5986633 - Thu Aug 13 04:17:54 UTC 2015
CPUs: 4
Total Memory: 3.858 GiB
Name: imc
ID: QZFV:WXGI:TPNT:3PZU:DD7D:RWKP:MNXY:R4PF:XGVS:XX2D:NEVD:BBJG
Debug mode (server): true
File Descriptors: 10
Goroutines: 27
System Time: 2015-08-18T19:42:37.145986554Z
EventsListeners: 0
Init SHA1:
Init Path: /usr/local/bin/docker
Docker Root Dir: /mnt/sda1/var/lib/docker
Labels:
 provider=virtualbox
```
uname -a:
``` Darwin Kernel Version 14.4.0  (Mac OSX)```

Provide additional environment details (AWS, VirtualBox, physical, etc.):
- docker-machine (Mac OSX, VirtualBox)

List the steps to reproduce the issue:
1. Create/save psFormat table, with (container) ID in any index, other than the first.  Here is my ${HOME}/.docker/config.json:
```json
{
  "psFormat": "table {{ .Names }}\\t{{ .ID }}\\t{{ .Status }}\\t{{ .Image }}"
}
``` 

2. Run docker ps --quiet

Describe the results you received:
All values from the psFormat table are displayed
```
>  docker ps --quiet
CONTAINER ID        NAMES               STATUS              IMAGE
c3674a480bea        high_bartik         Up 19 seconds       docker-registry:8080/roe
```

Describe the results you expected:
Only display numeric IDs, as the default behavior without psFormat table.  For example:
```
>  docker ps --quiet
c3674a480bea
```

Provide additional info you think is important:
BIG, big fan of the psFormat!  It would be nice to have this work with --quiet.  

Aside/wishlist/probably-a-separate-request:
It would be sweet to allow separate psFormat tables for list default (docker ps) and list all (docker ps --all).

**Comment by cpuguy83** [comment]
Hi, thanks for submitting.
This is already fixed in #15599

**Comment by paxet** [comment]
This strikes me again with docker 23.0.1
Totally reproducible as per @stayclassychicago instructions.

We are using it in bash scripts, to parse running containers
